bsp4j/src/main/java/ch/epfl/scala/bsp4j/BuildClient.java

```diff
@@ -0,0 +1,35 @@
+package ch.epfl.scala.bsp4j;
+
+import org.eclipse.lsp4j.jsonrpc.services.JsonNotification;
+import org.eclipse.lsp4j.jsonrpc.services.JsonRequest;
+
+import java.util.concurrent.CompletableFuture;
+
+public interface BuildClient {
+
+    @JsonNotification("build/showMessage")
+    void onBuildShowMessage(ShowMessageParams params);
+
+    @JsonNotification("build/logMessage")
+    void onBuildLogMessage(LogMessageParams params);
+
+    @JsonNotification("build/publishDiagnostics")
+    void onBuildPublishDiagnostics(PublishDiagnosticsParams params);
```

Comment:
Maybe we can still satisfy @jastice's request of keeping the same names as the spec by using the real event name but preprended by `on*` like `onPublishDiagnosticsParams`?

Reply:
the current `onBuildPublishDiagnostics` name matches the protocol method name instead of the paramter type.